test/ctia/test_helpers/core.clj

```diff
@@ -59,49 +60,58 @@
     (clear-property configurable-property))
   ;; Override any properties that are in the default properties file
   ;; yet are unsafe/undesirable for tests
-  (set-property "ctia.http.dev-reload" false)
-  (set-property "ctia.http.min-threads" 9)
-  (set-property "ctia.http.max-threads" 10)
-  (set-property "ctia.nrepl.enabled" false)
-  ;; Run tests
-  (f))
-
-(defn fixture-properties:atom-store [f]
-  ;; Set properties to enable the atom store
-  (with-properties ["ctia.store.type" "memory"]
+  (with-properties ["ctia.http.dev-reload" false
+                    "ctia.http.min-threads" 9
+                    "ctia.http.max-threads" 10
+                    "ctia.nrepl.enabled" false]
+    ;; run tests
     (f)))
 
-(defn fixture-properties:redis-store [f]
-  ;; May be overridden with ENV variables
-  (with-properties ["ctia.store.redis.enabled" true]
-    (f)))
+(defn fixture-property [prop val]
+  (fn [test]
+    (set-property prop val)
+    (test)))
 
-(defn fixture-properties:http-disabled [f]
-  (with-properties ["ctia.http.enabled" false]
+(defn fixture-properties:atom-store [f]
+  ;; Set properties to enable the atom store
+  (with-properties ["ctia.store.actor" "memory"
+                    "ctia.store.feedback" "memory"
+                    "ctia.store.campaign" "memory"
+                    "ctia.store.coa" "memory"
+                    "ctia.store.exploit-target" "memory"
+                    "ctia.store.identity" "memory"
+                    "ctia.store.incident" "memory"
+                    "ctia.store.indicator" "memory"
+                    "ctia.store.judgement" "memory"
+                    "ctia.store.sighting" "memory"
+                    "ctia.store.ttp" "memory"]
     (f)))
 
 (defn available-port []
   (with-open [sock (ServerSocket. 0)]
     (.getLocalPort sock)))
 
-(defn fixture-ctia [test]
-  ;; Start CTIA
-  ;; This starts the server on a random port, once for each
-  ;; (test), which _might_ briefly use up a lot of ports
-  (with-properties ["ctia.http.port" (available-port)]
-    (try
-      (init/start-ctia! :join? false
-                       :silent? true)
-      (test)
-      (finally
-        ;; explicitly stop the http-server
-        (http-server/stop!)
-        (e/shutdown!)
-        (rs/shutdown!)))))
-
-(def fixture-ctia-fast
-  (ct/join-fixtures [fixture-properties:http-disabled
-                     fixture-ctia]))
+(defn fixture-ctia
+  ([test] (fixture-ctia test true))
+  ([test enable-http?]
+   ;; Start CTIA
+   ;; This starts the server on a random port (if enabled)
+   (with-properties ["ctia.http.enabled" enable-http?
+                     "ctia.http.port" (if enable-http?
+                                        (available-port)
+                                        3000)]
```

Comment:
The available-port fn let's java pick a port.  The 3000 value is just a place holder for when http is disabled.